app/Metadata/MetadataServiceProvider.php

```diff
@@ -2,35 +2,44 @@
 
 declare(strict_types=1);
 
-namespace CultuurNet\UDB3\Silex\Metadata;
+namespace CultuurNet\UDB3\Metadata;
 
 use Broadway\Domain\Metadata;
 use Broadway\EventDispatcher\CallableEventDispatcher;
 use Broadway\EventSourcing\MetadataEnrichment\MetadataEnrichingEventStreamDecorator;
 use CultuurNet\UDB3\CommandHandling\ResqueCommandBus;
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
 use CultuurNet\UDB3\EventSourcing\LazyCallbackMetadataEnricher;
 use CultuurNet\UDB3\CommandHandling\ContextFactory;
-use CultuurNet\UDB3\Silex\Container\HybridContainerApplication;
-use Silex\Application;
-use Silex\ServiceProviderInterface;
 
-final class MetadataServiceProvider implements ServiceProviderInterface
+final class MetadataServiceProvider extends AbstractServiceProvider
 {
-    public function register(Application $app)
+    protected function getProvidedServiceNames(): array
     {
-        $app['context'] = null;
+        return [
+            'metadata_enricher',
+            'event_stream_metadata_enricher',
+            'command_bus_event_dispatcher',
+        ];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared('context', '');
 
-        $app['metadata_enricher'] = $app::share(
-            function (HybridContainerApplication $app) {
+        $container->addShared(
+            'metadata_enricher',
+            function () use ($container) {
                 return new LazyCallbackMetadataEnricher(
-                    function () use ($app) {
+                    function () use ($container) {
                         // Create a default context from application globals.
-                        $container = $app->getLeagueContainer();
                         $context = ContextFactory::createFromGlobals($container);
 
                         // Allow some processes to overwrite the context, like resque workers.
-                        if ($app['context'] instanceof Metadata) {
-                            $context = $app['context'];
+                        if ($container->get('context') instanceof Metadata) {
```

Comment:
Is there any code that still writes to `$app['context']` (or now `$container->addShared('context')`? I cannot find any. So I think this `if()` and the `$container->addShared('context', '');` can just be removed.

Reply:
+ If there was still a place that overwrote it using `$app['context'] = ...` that would not work as intended anymore at this point because it would write the `context` service into the Silex Application object, but the `context` in the PSR container would remain an empty string.